pkg/module_manager/module.go

```diff
@@ -942,6 +946,59 @@ func (mm *ModuleManager) RegisterModules() error {
 	return nil
 }
 
+// SyncModulesCR synchronize modules CR from current modules list to the cluster one
+func (mm *ModuleManager) SyncModulesCR(client klient.Client) error {
+	if mm.registerModulesGV == "" {
+		return nil
+	}
+
+	gvr, err := client.GroupVersionResource(v1alpha1.ModuleGVK.GroupVersion().String(), v1alpha1.ModuleGVK.Kind)
+	if err != nil {
+		return err
+	}
+
+	cctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
```

Comment:
```suggestion
	cctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```
It seems like timeout is too low.